lib/libzrepl/data_conn.c

```diff
@@ -670,10 +680,32 @@ uzfs_zvol_rebuild_dw_replica(void *arg)
 			 * one of them might have changed rebuild state
 			 */
 			if (uzfs_zvol_get_rebuild_status(zinfo->main_zv) !=
-			    ZVOL_REBUILDING_AFS)
+			    ZVOL_REBUILDING_AFS) {
 				uzfs_zvol_set_rebuild_status(zinfo->main_zv,
 				    ZVOL_REBUILDING_AFS);
+				/*
+				 * Lets ask io_receiver thread to flush
+				 * all outstanding IOs in taskq
+				 */
+				zinfo->quiesce_done = 0;
+				zinfo->quiesce_requested = 1;
+			}
 			mutex_exit(&zinfo->main_zv->rebuild_mtx);
+			/*
+			 * Wait for all outstanding IOs to be flushed
+			 * to disk before making further progress
+			 */
+			while (1) {
+				if (zinfo->quiesce_done ||
+				    !taskq_check_active_ios(
+				    zinfo->uzfs_zvol_taskq)) {
+					zinfo->quiesce_done = 1;
```

Comment:
do we need to set zinfo->quiesce_done to one? if condition already make sure of it.

Reply:
never mind, saw the || condition. looks ok.

Reply:
ok